server/src/main/java/org/opensearch/search/SearchService.java

```diff
@@ -272,6 +272,13 @@ public class SearchService extends AbstractLifecycleComponent implements IndexEv
         Property.NodeScope
     );
 
+    public static final Setting<Boolean> FILTER_REWRITE_SETTING = Setting.boolSetting(
+        "search.filter_rewrite.enabled",
+        true,
+        Property.Dynamic,
+        Property.NodeScope
+    );
+
```

Comment:
I am wondering if there is better name for this setting? Or can we have the number of buckets into this setting to use single setting? Setting to 0 disables the use of rewrite optimization altogether

Reply:
Discussed offline and updated to this approach. This setting now is an integer, 0 means disable, >0 values will be treated as the max filters threshold to do the optimization.